models/DataObject/QuantityValue/Service.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Model\DataObject\QuantityValue;
+
+use Pimcore\Model\Translation;
+
+class Service
+{
+    public static function importDefinitionFromJson(string $json, $deleteExisting = true): bool
+    {
+        try {
+            $unitsArray = json_decode($json, true);
+            $baseUnits = array_column($unitsArray, 'baseunit');
+            $units = []; //array of units to be imported;
+            foreach ($unitsArray as $unitArray) {
+                if (($unit = Unit::getById($unitArray['id'])) && $deleteExisting) {
+                    $unit->delete();
+                }
+                $unit = Unit::create($unitArray);
```

Comment:
```suggestion
                foreach ($unitArray as $key => $value){
                    if(empty($value)){
                        unset($unitArray[$key]);
                    }
                }

                $unit = Unit::create($unitArray);
```

Reply:
You have to eliminate the empty values since this will call an error.
For example, `setFactor()` only accepts `float`.

https://github.com/pimcore/pimcore/blob/c91db766bee2d98637615fa5746166bee78fca9e/models/DataObject/QuantityValue/Unit.php#L189

Reply:
Since you provide a "do not override" option with `$deleteExisting` you have to merge the existing entry with the imported one OR just skip the import if the unit (with the ID already exists).

ATM every import overrides the existing unit. Even if you provide the `--no-delete` flat at the command.
Maybe renaming `delete` to `override` would it more clear. - WDYT?

Or do I understand something wrong?

Reply:
```
            foreach ($unitsArray as $unitArray) {
                if ($unit = Unit::getById($unitArray['id'])) {
                    if ($deleteExisting) { // delete the existing entry
                        $unit->delete();
                    }
                    else { //skip the import if delete flag is not set
                        continue;
                    }
                }

  ```
  The idea of merging two units with same identifier is not the desired behavior, so I think it will be wise to choose to skip the import in case, if the unit definition  for that entry already exists. What do you say?

Reply:
See https://github.com/pimcore/pimcore/pull/14295 `setFactor()` should accept null.

Reply:
> The idea of merging two units with same identifier is not the desired behavior, so I think it will be wise to choose to skip the import in case, if the unit definition for that entry already exists. What do you say?

I personally would prefer to have the option. If I pass the `--override` flag, the existing entry should be replaced with the new one. If I don't pass an option then you could just skip the import for this definition. So `--no-override` would be the default behavior. 

Then there would be no "merge" ....

WDYT

Reply:
Okay, now it supports the override as suggested in this code, also updated the naming of the flags, so it would be more verbose. Thanks!

Reply:
Now, it has the changes as suggested, thanks for the review.  Name of the flag changed to override, as it is more clear to understand, plus, no override is the default behavior.

Reply:
nice that sounds solid 👍 will have another look at it this week 🍭